src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -130,9 +130,10 @@ public void onResolveFailed(NsdServiceInfo serviceInfo, int errorCode) {
             Log.w(
                 TAG,
                 "Failed to resolve service '" + serviceInfo.getServiceName() + "': " + errorCode);
+            String serviceType = serviceInfo.getServiceType();
             chipMdnsCallback.handleServiceResolve(
                 serviceInfo.getServiceName(),
-                serviceInfo.getServiceType(),
+                serviceType.startsWith(".") ? serviceType.substring(1) : serviceType,
```

Comment:
You could alternatively also avoid this substring by doing this:
```
                // Use the target service info since the resolved service info sometimes appends a "." at the front
                // likely because it is trying to strip the service name out of it and something is missed.
                // The target service info service type should be effectively the same as the resolved service info.
                NsdServiceFinderAndResolver.this.targetServiceInfo.getServiceType(),
```

It might be helpful to add a comment here too indicating why we're doing this
Thanks for adding this, i was just about to push a very similar change here to fix this as well

Reply:
Modified. Thanks.